app/lib/frontend/name_tracker.dart

```diff
@@ -71,14 +71,18 @@ class NameTracker {
 class NameTrackerUpdater {
   final DatastoreDB _db;
   DateTime _lastTs;
+  Completer _sleepCompleter;
+  Timer _sleepTimer;
+  bool _stopped = false;
 
   NameTrackerUpdater(this._db);
 
-  // Note, this method never returns.
+  /// The returned future completes after the `stop` method is called.
   Future startNameTrackerUpdates() async {
```

Comment:
This should throw a `StateError` if it has already been stopped... or at-least that would be good :)

Note: this is not important.